src/core/server/legacy_compat/legacy_platform_proxifier.ts

```diff
@@ -18,16 +18,29 @@
  */
 
 import { EventEmitter } from 'events';
-import { IncomingMessage, ServerResponse } from 'http';
 import { Server } from 'net';
 
+import { Server as HapiServer, ServerOptions as HapiServerOptions } from 'hapi-latest';
+import { Env } from '../config';
 import { Logger } from '../logging';
 import { Root } from '../root';
 
+interface ConnectionInfo {
+  server: HapiServer;
+  options: HapiServerOptions;
+}
+
 /**
  * List of the server events to be forwarded to the legacy platform.
  */
-const ServerEventsToForward = ['listening', 'error', 'clientError', 'connection'];
+const ServerEventsToForward = [
```

Comment:
__note:__ I added few more events (e.g. `upgrade`) to support socket.io functionality that Canvas uses.